src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs

```diff
@@ -346,17 +346,26 @@ public void Http3ConnectionClosed(string connectionId, long highestOpenedStreamI
 
         public void Http3StreamAbort(string traceIdentifier, Http3ErrorCode error, ConnectionAbortedException abortReason)
         {
-            _http3StreamAbort(_logger, traceIdentifier, error, abortReason);
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                _http3StreamAbort(_logger, traceIdentifier, Http3Formatting.ToFormattedErrorCode(error), abortReason);
```

Comment:
Nit: Are the ToFormatted methods expensive enough to warrant an extra IsEnabled check?

Reply:
For known values, probably not. But if there is an unknown value a new string will get allocated.